packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1079,6 +1079,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         this.summaryCollection = new SummaryCollection(this.deltaManager, this.logger);
 
+        const { attachState, pendingLocalState } = this.context;
+        this.dirtyContainer = attachState !== AttachState.Attached
+            || (pendingLocalState as IPendingLocalState)?.pendingStates.length > 0;
```

Comment:
I think it's worth opening an issue to follow up here to not have such cast in the future

Reply:
I combined coupe of items we discussed around pending states and pending state manager here: https://github.com/microsoft/FluidFramework/issues/8893